flow/layers/color_filter_layer.cc

```diff
@@ -54,18 +62,23 @@ void ColorFilterLayer::Preroll(PrerollContext* context) {
   // modulation so in that case we can apply the opacity on behalf of our
   // ancestors - otherwise we can apply no attributes.
   if (filter_) {
+    set_paint_bounds(child_bounds);
+
     context->renderable_state_flags =
         filter_->can_commute_with_opacity()
             ? LayerStateStack::kCallerCanApplyOpacity
             : 0;
+  } else {
+    // else - we can apply whatever our children can apply.
+    set_paint_bounds(child_bounds);
```

Comment:
This line is not conditional, why is it applied in both halves of the if?

Reply:
fixed